CMakeLists.txt

```diff
@@ -74,20 +78,25 @@ if (NOT MSVC)
   set(CMAKE_CXX_FLAGS_RELEASE "-O3 -DNDEBUG")
 endif ()
 message(STATUS "compiler ${CMAKE_CXX_COMPILER_ID} version ${CMAKE_CXX_COMPILER_VERSION}")
+set (WARN_FLAGS "")
 if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
   set (CXX11_FLAGS "-std=c++11")
-  set (WARN_FLAGS "-Weverything -Wno-c++98-compat -Wno-missing-prototypes")
-  set (WARN_FLAGS "${WARN_FLAGS} -Wno-c++98-compat-pedantic -Wno-padded")
-  set (WARN_FLAGS "${WARN_FLAGS} -Wno-covered-switch-default")
-  set (WARN_FLAGS "${WARN_FLAGS} -Wno-missing-noreturn -Wno-unknown-pragmas")
-  set (WARN_FLAGS "${WARN_FLAGS} -Wno-gnu-zero-variadic-macro-arguments")
-  set (WARN_FLAGS "${WARN_FLAGS} -Wconversion")
-  if (CMAKE_HOST_APPLE AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "11.0")
-    set (WARN_FLAGS "${WARN_FLAGS} -Wno-c++2a-compat")
+  if (ENABLE_CPP_WARNINGS)
```

Comment:
Use 
if (STOP_BUILD_ON_WARNING) 
  set (WARN_FLAGS "${WARN_FLAGS) -Werror")
endif ()

We absolutely should get the warnings even if they are building a non-dev build.